src/components/Dashboard.tsx

```diff
@@ -66,9 +69,16 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
               {template?.namePlural || `${name} ${strings.LABEL_APPLICATIONS}`}
             </a>
             <Icon name="chevron right" />
+            {loadedFiltersCount !== filters.length && <LoadingSmall />}
           </Label>
-          {filters.map((filter) => (
-            <FilterComponent key={filter.id} template={template} filter={filter} />
+          {filters.map((filter, i: number, filters) => (
+            <FilterComponent
+              key={filter.id}
+              template={template}
+              filter={filter}
+              setFiltersCount={setLoadedFiltersCount}
+              filters={filters}
+            />
```

Comment:
```suggestion
          {filters.map((filter) => (
            <FilterComponent
              key={filter.id}
              template={template}
              filter={filter}
              setFiltersCount={setLoadedFiltersCount}
            />
```
Not using index or filters now.